pkg/core/tekton/tekton.go

```diff
@@ -0,0 +1,405 @@
+package tekton
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/pkg/apis"
+
+	"github.com/fuseml/fuseml-core/gen/codeset"
+	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/core/tekton/builder"
+)
+
+// WorkflowBackend implements the FuseML WorkflowBackend interface for tekton
+type WorkflowBackend struct {
```

Comment:
Ah, yes, that makes sense. It will be referenced as `tekton.WorkflowBackend` from outside the package, which gives it the "tekton" specificity that I was looking for.